### DISCUSSION TITLE
Can DGS work alongside with Kotlin's coroutines?

### DISCUSSION THREAD

**Discussion by padjiwib-intuit** [Netflix/dgs-framework#436] [Q&A]
Starting Spring 5.2, Spring supports Kotlin's coroutines as its first class citizen. However, we are not sure whether this coroutine support work well alongside with DGS framework. Could you please help advise?

Thanks!

**Comment by paulbakker** [comment]
I have no experience with coroutines (which I agree is strange, because Kotlin is pretty much all I do...), so I can't comment on this. 
I do plan to look into this further, but I don't currently have any answers ready.

Would love to hear from someone more familiar with it.

**Reply by padjiwib-intuit** [reply]
Thanks for the prompt response. Do you have certain timeline by which you can help take a look and come up with, at least, an answer whether it is recommended to mix DGS with Kotlin's coroutines? I don't expect things to be working perfectly, but having been able to understand potential issues/bottlenecks ahead of time can help us decide whether we should go for coroutine approach or other. And of course if we can contribute to make the framework more robust, we would definitely be up for that.

Thanks again in advance!
